### DISCUSSION TITLE
Safer `.env*` integration

### DISCUSSION THREAD

**Discussion by hbarcelos** [vercel/next.js#24229] [Ideas]
### Describe the feature you'd like to request

When using `.env*` files, it's usually a good practice to define a `.env.example` file with the expected environment variables and not version control other `.env*` files to avoid exposing potential sensitive data when pushing to a public repository.

Specially in the context of a team, for example, if one of the members adds a variable to `.env.example` and another team member inadvertently updates the code and tries to run it, they might get into trouble because of a missing variable.

### Describe the solution you'd like

Next should incorporate a feature like [`dotenv-safe`](https://github.com/rolodato/dotenv-safe), which warns the user in case of missing variables by throwing an error.

This should probably be enabled by default and disabled through a config option, but maybe for retro-compatibility it could be the other way around. 

### Describe alternatives you've considered

The alternatives currently require conscious effort to:
1. Always check `.env.example` for changes.
2. Make assertions in the code looking for environment variables.

While they could solve the problem, is yet another thing developers must have in mind, which is sub-optimal.

**Comment by obedm503** [comment]
Having dotenv-safe integration would help teams with CI and CD environments specially with new deployments that may not yet have defined certain environment variables.
